src/core/server/saved_objects/service/lib/update_objects_spaces.ts

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { BulkOperationContainer, MultiGetOperation } from '@elastic/elasticsearch/api/types';
+import intersection from 'lodash/intersection';
+
+import type { ISavedObjectTypeRegistry } from '../../saved_objects_type_registry';
+import type { SavedObjectsRawDocSource, SavedObjectsSerializer } from '../../serialization';
+import type {
+  MutatingOperationRefreshSetting,
+  SavedObjectError,
+  SavedObjectsBaseOptions,
+} from '../../types';
+import type { DecoratedError } from './errors';
+import { SavedObjectsErrorHelpers } from './errors';
+import {
+  getBulkOperationError,
+  getExpectedVersionProperties,
+  rawDocExistsInNamespace,
+} from './internal_utils';
+import { DEFAULT_REFRESH_SETTING } from './repository';
+import type { RepositoryEsClient } from './repository_es_client';
+
+/**
+ * An object that should have its spaces updated.
+ *
+ * @public
```

Comment:
nit: `@public` is not required. The act of exporting this out of `src/core/server` is enough to mark this as public